plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeConnectorTest.java

```diff
@@ -367,4 +367,35 @@ public void testSelectInformationSchemaColumns()
     {
         // TODO https://github.com/trinodb/trino/issues/21157 Enable this test after fixing the timeout issue
     }
+
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        switch (setup.newColumnType()) {
+            case "smallint":
+            case "integer":
+                return Optional.empty();
+        }
+
+        if (setup.sourceColumnType().startsWith("row") || setup.newColumnType().startsWith("array")) {
+            return Optional.of(setup.asUnsupported());
+        }
+
+        switch ("%s -> %s".formatted(setup.sourceColumnType(), setup.newColumnType())) {
+            case "decimal(5,3) -> decimal(5,2)":
+            case "varchar -> char(20)":
+            case "time(6) -> time(3)":
+            case "timestamp(6) -> timestamp(3)":
+                return Optional.of(setup.asUnsupported());
+        }
+
+        return Optional.of(setup);
+    }
+
+    @Override
+    protected void verifySetColumnTypeFailurePermissible(Throwable e)
+    {
+        assertThat(e).hasMessageMatching("[\\s\\S]*(SQL compilation error: cannot change column COL from type .* to .*" +
+                "|This connector does not support setting field types)[\\s\\S]*");
```

Comment:
The failure message shouldn't have "This connector does not support setting field types".